code/modules/admin/holder2.dm

```diff
@@ -1,13 +1,25 @@
 GLOBAL_LIST_EMPTY(admin_datums)
 GLOBAL_PROTECT(admin_datums) // This is protected because we dont want people making their own admin ranks, for obvious reasons
 
+GLOBAL_VAR_INIT(href_token, GenerateToken())
+GLOBAL_PROTECT(href_token)
+
+/proc/GenerateToken()
+	. = ""
+	for(var/I in 1 to 32)
+		. += "[rand(10)]"
+
 /datum/admins
 	var/rank = "Temporary Admin"
 	var/client/owner
 	/// Bitflag containing the current rights this admin holder is assigned to
 	var/rights = 0
 	var/fakekey
 	var/big_brother	= FALSE
+
+	/// Unique-to-session randomly generated token given to each admin to help add detail to logs on admin interactions with hrefs
+	var/href_token
```

Comment:
Is it ever possible 2 admins have the same href token?

Reply:
It is a possibility, although exceedingly improbable, this is the token generation code:

```
/proc/GenerateToken()
	. = ""
	for(var/I in 1 to 32)
		. += "[rand(10)]"
```

Reply:
Example output:
![image](https://github.com/ParadiseSS13/Paradise/assets/12197162/267685af-148a-4d50-9d72-4e22c2a7bf3d)

Reply:
(that's 32^11 possible combos, or 36028797000000000)